### DISCUSSION TITLE
Panel background color?

### DISCUSSION THREAD

**Discussion by FedeClaudi** [Textualize/rich#405] [Q&A] [answered]
Hi!

I was wandering if there's a way to set a panel's background color?

I can change the panel's text and edge colors (e.g. see screnshot from [knowledgebase](https://github.com/FedeClaudi/knowldege_base)), but I haven't found a way to "fill in" the whole panel with color, any suggestions?

Somewhat related, is there an options to have a "transparent" color?  Sometimes I use visual elements like panels to make sure things are aligned correctly but I don't want to see the panel. Currently I'm hard coding the panel's edge to match my terminal color but it doesn't sound like a long term solution.

![image](https://user-images.githubusercontent.com/17436313/97094464-d4e47680-164c-11eb-80e0-7d270f3b541e.png)

Thanks Will, Rich is so much fun :D

**Comment by willmcgugan** [comment, marked as answer]
You can set a background color with the style argument on Panel. For example `Panel("Hello", style="on blue")`

There's no transparent Panel per se. But you could use [Padding](https://rich.readthedocs.io/en/latest/reference/padding.html) to create a one cell border around something. For instance `Padding(renderable, 1)` 

Glad you like Rich. You've done some awesome stuff with it lately!

**Reply by FedeClaudi** [reply]
thanks!
the `on color` option fills in the whole segment taken up by the panel:

<img width="632" alt="image" src="https://user-images.githubusercontent.com/17436313/97106340-e6636800-16b8-11eb-9b21-b87499b54b72.png">

I think that makes sense, but it's not what I had in mind, I was trying to get the panels to look like 
<img width="181" alt="image" src="https://user-images.githubusercontent.com/17436313/97106368-009d4600-16b9-11eb-85ec-c364e7c94776.png">

so only coloring within the edges. I guess that might be very hard to achieve though?

**Reply by willmcgugan** [reply]
> so only coloring within the edges. I guess that might be very hard to achieve though?

Impossible on a terminal I'm afraid!

**Reply by FedeClaudi** [reply]
thought so, fair enough. Thanks for help anyway :)
